src/ipyaladin/widget.py

```diff
@@ -262,6 +273,93 @@ def selected_objects(self) -> List[Table]:
             catalogs.append(Table(objects_data))
         return catalogs
 
+    @property
+    def selected_regions(self) -> SupportedSelectionRegion:
+        """The regions selected by the user in a given session.
+
+        Returns
+        -------
+        _______
+        List[`~regions.CircleSkyRegion`, `~regions.RectangleSkyRegion`]
+            An astropy region object representing the region selected by the user.
+
+        """
+        if Region is None:
+            raise ModuleNotFoundError(
+                "To read regions objects, you need to install the regions library with "
+                "'pip install regions'."
+            )
+
+        selected_regions = []
+        for region in self._selected_regions:
+            region_type = region.get("type", None)
+
+            if region_type == "circle":
+                startCoo = region.get("startCoo", None)
+                endCoo = region.get("endCoo", None)
+                r2 = (endCoo["ra"] - startCoo["ra"]) * (
+                    endCoo["ra"] - startCoo["ra"]
+                ) + (endCoo["dec"] - startCoo["dec"]) * (
+                    endCoo["dec"] - startCoo["dec"]
+                )
+                r = math.sqrt(r2)
+
+                center = SkyCoord(
+                    startCoo["ra"], startCoo["dec"], unit="deg", frame="icrs"
+                )
+
+                selected_regions.append(CircleSkyRegion(center, radius=r * u.deg))
+
+            elif region_type == "rect":
+                startCoo = region.get("startCoo", None)
+                endCoo = region.get("endCoo", None)
+                w = abs(endCoo["ra"] - startCoo["ra"])
+                h = abs(endCoo["dec"] - startCoo["dec"])
+                x = (endCoo["ra"] + startCoo["ra"]) / 2
+                y = (endCoo["dec"] + startCoo["dec"]) / 2
```

Comment:
For the same reasons as with handling the coords for the circle, we'll need to do this a bit differently.  

I'm actually realizing I have some questions about the semantics of the `RectangleSkyRegion` and I don't think we want to deal with possible rotation angles if north isn't straight up in the UI, so I'm thinking it might be better to first compute the corners in screen coords, convert those to world coords, then make a PolygonSkyRegion.  

The actual polygon case below should be working well since each vertex is converted directly from screen to world with no distances on the sphere being needed.

Reply:
We spoke of this on friday, and came to the conclusion that handling `RectangleSkyRegion`s as `PolygonSkyRegion`s was the safest approach. This change was implemented